### PR TITLE
Improved solve_univariate_inequality function

### PR DESCRIPTION
**Pull request by arihantparsoya**
This is extension of #11109 . function `_real_conditions()` is better implemented
#### Sample Program

``` python
>>> from sympy import *
>>>from sympy.abc import x
>>> solveset(x + sqrt(x**2-5) > 0, x, S.Reals)
(sqrt(5), oo)
>>> solveset(x + sqrt(x**2-5) < 0, x, S.Reals)
(-oo, -sqrt(5))
>>> solveset(2 + sqrt(x - 4) > 0, x, S.Reals)
(4, oo)
```

### CODE REVIEW

**Comment by parsoyaarihant** [comment]
@Shekharrajak can you review this?

**Comment by parsoyaarihant** [comment]
@smichr  , can you review this?

File: sympy/solvers/inequalities.py [hunk 1]
```diff
@@ -481,11 +483,32 @@ def valid(x):
...
+            nsol_sets = _real_conditions(expr,gen)
+        cv = Intersection(*nsol_sets).subs(gen,_gen)
         rv = Union(*sol_sets).subs(gen, _gen)
-
+        rv = Union(rv, cv)
     return rv if not relational else rv.as_relational(_gen)
 
 
+def _real_conditions(expr, gen):
+    from sympy.core import Mul, Pow
```
Comment (sampadsaha5) [inline sympy/solvers/inequalities.py]: Please add all the imports in the top of the file only. It is a good practice except for the case of cyclic imports.

Comment (parsoyaarihant) [inline sympy/solvers/inequalities.py]: @sampadsaha5 , thanks. Can you give your review on the whole PR so I can make changes if any?

**Comment by kshitij10496** [comment]
@parsoyaarihant 
I think [PR#11458](https://github.com/sympy/sympy/pull/11458) addresses these issues.

Also, 
```python 
>>> solveset(2 + sqrt(x - 4) > 0, x, S.Reals)
(4, oo) # should be
 ``` 

This should be : `[4, 8) U (8, oo)`

**Pull request closed without merging**